src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -263,7 +265,9 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
             this.showCopyStatusMessage();
           });
         }
-      }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
+        this.sessionEditFormModel.isCopying = false;
+      }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); })
+      .catch(() => this.sessionEditFormModel.isCopying = false);
```

Comment:
Why not use `.pipe(finalize(...))`. That way, you only need to set the `isCopying` in one place. See the other places with the loading spinner

Reply:
I tried to use `.pipe(finalize(...))` but I cannot find a place to use it. Property 'pipe' does not exist on type 'Promise<void>' or 'Subscription'.

Reply:
The Promise's equivalance of `finalize` is `.finally()`. You can try `.finally(() => this.sessionEditFormModel.isCopying = false)`, instead of repeating the same code in 3 places.

Reply:
I tried using `finally` and remove the repeated code, but doing so will make the spinner to stop prematurely while the backend is still copying.
```
modalRef.result.then((result: CopySessionModalResult) => {
        this.failedToCopySessions = {};
        const requestList: Observable<FeedbackSession>[] = this.createSessionCopyRequestsFromModal(
            result, this.courseId, this.feedbackSessionName);
        if (requestList.length === 1) {
          requestList[0].subscribe((createdSession: FeedbackSession) => {
            if (Object.keys(this.failedToCopySessions).length === 0) {
              this.navigationService.navigateWithSuccessMessage(this.router,
                  '/web/instructor/sessions/edit',
                  'The feedback session has been copied. Please modify settings/questions as necessary.',
                  { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
            } else {
              this.statusMessageService.showErrorToast(this.getCopyErrorMessage());
            }
          }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
        }
        if (requestList.length > 1) {
          forkJoin(requestList)
           .subscribe(() => {
             this.showCopyStatusMessage();
           });
        }
      }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); })
      .finally(() => this.sessionEditFormModel.isCopying = false);
```
The result of such changes looks like this
![copy20210623_165354](https://user-images.githubusercontent.com/53338059/123067650-be90f380-d443-11eb-998c-d964ec8c5ee7.gif)

Reply:
Yeah I don't think there's an easy way to do it all in one place for now, since the modal closes before the request returns.